src/validator.ts

```diff
@@ -60,6 +60,17 @@ const isNotAbstractClass = (node: Class<'Linked'>) =>
 export const validations = (environment: Environment) => {
   const { parentOf, firstAncestorOfKind, resolveTarget } = tools(environment)
 
+  // Can something like <Self<'Linked'> isNotPresentIn 'Program' be done? (typescript interprets it like a cast :( )
```

Comment:
Sadly, no. At least not in any way that I know of. You can go ahead and remove the comment and merge.